pyproject.toml

```diff
@@ -46,10 +45,16 @@ cli = [
     "rich>=10,<14",
 ]
 http2 = [
-    "h2>=3,<5",
+    "httpcore[http2]",
 ]
 socks = [
-    "socksio==1.*",
+    "httpcore[socks]",
+]
+asyncio = [
+    "httpcore[asyncio]"
+]
+trio = [
+    "httpcore[trio]"
 ]
```

Comment:
This is interesting... currently we're including the `httpcore` dependencies explicitly.

Eg. `pip install httpx[socks]` uses the dependency `socksio==1.*` instead of the dependency `httpcore[socks]`.

The second style makes more sense, because that's what we're *actually* trying to effect... but I'm a bit hesitant that *some* Python install tooling might incorrectly fail to install `httpcore[socks]` as an optional dependency when the `httpcore` dependancy is already included. Is my hesitancy well founded here, or are we okay to use this new style throughout?